### PR TITLE
Set up repo and create utilities library

### PR DESCRIPTION
**Pull request by india-kerle**
This pull request starts to address #9 - the DataGetter class does NOT contain all the methods used to load different datasets. 

This pull request does the following:

1. It adds structure to the repo where `tutorials` will contain the tutorials for taltech hackweek 2023. It currently contains an illustrative directory called `text_analysis/` that contains an empty readme. 
2. It creates a `utilities` library that currently contains a class to load data either from s3 or locally that is relevant across tutorials. you can just `pip install -r requirements.txt` at the root of the repository to install the utilities library. 

To make sure the code runs:

```
conda create -n dap_taltech python=3.9
pip install git+https://github.com/nestauk/dap_taltech.git@repo_setup
```

Then:
```
from dap_taltech.utils.data_getters import DataGetter

dg = DataGetter(local=True)
estonian_patents = dg.get_estonian_patents()
```

### CODE REVIEW

**Review by ampudia19 — APPROVED** [review]
All looks good.

File: dap_taltech/utils/data_getters.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        If True, load data from local data directory. Else, download data from open dap-taltech s3 bucket.
+
+    """
+
+    def __init__(
+        self,
+        verbose=True,
+        local=True
+    ):
+        self.verbose = verbose,
```
Comment (ampudia19) [inline dap_taltech/utils/data_getters.py]: Doesn't the comma instantiate this is as a tuple?

File: dap_taltech/utils/data_getters.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        This data was collected using Google Patents. A patent is considered Estonian
+            if at least one inventor is Estonian. Patents were de-duplicated based on
+            family ID.
+
+        The data includes information such as:
+            - patent_id: unique identifier for each patent
+            - family_id: unique identifier for each patent family
+            - title_localized: title of the patent
+            - abstract_localized: abstract of the patent
+        """
```
Comment (ampudia19) [inline dap_taltech/utils/data_getters.py:75]: These docstrings are missing return sections, although this is v redundant with a getter.

File: dap_taltech/utils/data_getters.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        else:
+            logger.setLevel('ERROR')
+        if self.local:
+            self.data_dir = os.path.join(PROJECT_DIR, PUBLIC_DATA_FOLDER_NAME)
+            logger.info(f'Loading data from {self.data_dir}/')
+            if not os.path.exists(self.data_dir):
+                logger.warning(
+                    "Neccessary data files are not downloaded. Downloading neccessary files..."
+                )
+                os.system(f'aws s3 sync s3://{BUCKET_NAME}/data {self.data_dir}')
```
Comment (ampudia19) [inline dap_taltech/utils/data_getters.py:55]: Interesting to use cmd calls. Isn't the risk to have environment-specific issues pop up here? Isn't it easier to use Python native libraries instead (boto3)

**Review by ampudia19 — APPROVED** [review]
Good :)

File: dap_taltech/utils/data_getters.py (hunk 1 above)
Comment (india-kerle) [reply dap_taltech/utils/data_getters.py]: omg

**Pull request merged**